nmosnode/aggregator.py

```diff
@@ -317,15 +331,57 @@ def _get_api_href(self):
             api_href = self.mdnsbridge.getHref(LEGACY_REG_MDNSTYPE, None, AGGREGATOR_APIVERSION, protocol)
         return api_href
 
-    # Handle sending all requests to the Registration API, and searching for a new 'aggregator' if one fails
+    def register_auth_client(self, node_object):
+        """Function for Registering OAuth client with Auth Server and instantiating OAuth Client class"""
+        client_name = node_object['data']['description']
+        client_uri = node_object['data']['href']
+
+        if OAUTH_MODE is True and self.auth_registrar is None:
+            self.auth_registrar = self._auth_register(
+                client_name=client_name,
+                client_uri=client_uri
+            )
+        if self._registered['auth_client_registered']:
+            self.auth_registry = auth_registry
```

Comment:
I see this now. Having a lowercase variable initialised in global scope feels quite wrong. Is this really a constant that could be uppercased, or is there any other way it could be defined? It feels like magic at the moment...

Reply:
It should probably be upper-cased, I wasn't sure on the syntax of global objects as opposed to literals. 
This is set in the global scope as it is required in `api.py` both for initialisation with the flask instance and general use. The alternative is to define it in the `nodefacadeservice.py` if thats preferable... I imagine it is now I've thought about it